components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2751,8 +2701,10 @@ where
         assert!(!self.fsm.peer.is_handling_snapshot());
 
         // No need to wait for the apply anymore.
-        if self.fsm.unsafe_recovery_target_commit_index.is_some() {
-            self.finish_unsafe_recovery_wait_apply();
+        if self.fsm.peer.unsafe_recovery_state.is_some() {
```

Comment:
What if the message is still in queue when the peer is about to destroy itself? The refcount may not go to zero in the end.

Reply:
You are talking about the value stored inside the counter, not the ref count of the counter itself right? If the case happens, TiKV relies on PD to resend the plan again after a "timeout"